lib/core_extensions.rb

```diff
@@ -51,6 +59,19 @@ def before_destroy(record)
         false
       end
     end
+
+    def check_permissions(record, klass, klass_name, what)
+      association = record.class.reflect_on_association(klass.to_sym).klass
+      association = Host if association == Host::Managed
+      permission = "view_#{klass_name}"
```

Comment:
Reusing the authorized_associations_permission_name helper from cdb7071 might be useful?  It's also proposed to be improved in #2619.